Add Entry: CSS cursor (#640)
Codecademy/docs: content/css/concepts/cursor/cursor.md

---
Title: 'cursor'
Description: 'Sets the appearance of the mouse cursor when hovering over an element.'
Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'UI'
CatalogContent:
  - 'learn-css'
  - 'learn-intermediate-css'
---

The `cursor` property sets the appearance of the mouse cursor when hovering over an element. The appearance can be set by specifying the type of cursor, or by loading a specific icon. When using an icon, fallback images can be used, ending with a mandatory keyword fallback.

These are common keywords used to specify the type of cursor to display:

> ![Cursors](https://raw.githubusercontent.com/Codecademy/docs/main/media/cursors-1.png)

Author: I don't think it would wok with the current graphic. I could replace it with a bulleted list, but I thought the images were more helpful.

> ![Cursors](https://raw.githubusercontent.com/Codecademy/docs/main/media/cursors-1.png)

Reviewer: Maybe we could combine these two ideas into a markdown table?

Example:

| Name | Image | Description |
| ------ | ------ | ----------- |
| Default | ![default-cursor alt-text](...) | Used on most platforms; looks like an arrow. |
| Progress | ![progress-cursor alt-text](...) | Running in the background; user actions allowed. |
| Wait | ![wait-cursor alt-text](...) | Running in the background; user actions not allowed. |

> ![Cursors](https://raw.githubusercontent.com/Codecademy/docs/main/media/cursors-1.png)

Reviewer 2: Maybe instead of explaining each one, we can group certain ones together depending on their purpose?

Like which ones are used for selections, text, etc. Any rather than having those explanations in the images, we can include some bullet points underneath the image 🤔

> ![Cursors](https://raw.githubusercontent.com/Codecademy/docs/main/media/cursors-1.png)

Reviewer 3: I like the markdown table, too. To incorporate some of what @yangc95, the keywords could be grouped by category within the table. Similar to how they appear to be listed by category in the image.

> ![Cursors](https://raw.githubusercontent.com/Codecademy/docs/main/media/cursors-1.png)

Author: I'll try the markdown table approach. However, it will spam the media folder with a bunch of images for one entry. Anyone see this as a problem?

> ![Cursors](https://raw.githubusercontent.com/Codecademy/docs/main/media/cursors-1.png)

Reviewer 2: 😮  the image size is rather small so I don't think it would be an issue

---
Title: 'cursor'
Description: 'Sets the appearance of the mouse cursor when hovering over an element.'
Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Style'
  - 'UI'
CatalogContent:
  - 'learn-css'
  - 'learn-intermediate-css'
---

The `cursor` property sets the appearance of the mouse cursor when hovering over an element. The appearance can be set by specifying the type of cursor, or by loading a specific icon. When using an icon, fallback images can be used, ending with a mandatory keyword fallback.

These are some common keywords used to specify the type of cursor to display:

![Cursors](https://raw.githubusercontent.com/Codecademy/docs/main/media/cursors-1.png)

> ## Syntax

Reviewer: This is me, personally, but I would put the image + supporting text between the "Syntax" and "Example" sections.

> Without a URL:

Reviewer: ```suggestion
The `cursor` property can be set a specific `keyword` value:
```

> Without a URL:
> 

Reviewer: ```suggestion
```

```pseudo
cursor: keyword;
```

> With one or more URLs:

Reviewer: ```suggestion
It can also accept one or more URLs:
```

```pseudo

> cursor: url(cursor_image),url(fallback_cursor_image), keyword;

Reviewer: ```suggestion
cursor: url(cursor_image), url(fallback_cursor_image), keyword;
```

```

> Where `cursor_image` is the URL of the primary image for the cursor, `fallback_cursor_image` is optional. It is the URL to load if the prior image fails. There can be any number of fallback images. `keyword` is a mandatory cursor specifier in case all the image URLs fail to load.

Reviewer: ```suggestion
If the primary `cursor_image` fails to load, an optional `fallback_cursor_image` can be applied instead. However, in case all the image URLs fail to load, the mandatory `keyword` will be used as a last resort.
```

## Example

> 

Reviewer: ```suggestion

In the example below, a `pointer` hand will appear whenever the cursor is hovered over the [paragraph](https://www.codecademy.com/resources/docs/html/paragraphs):

```

---

The `cursor` property sets the appearance of the mouse cursor when hovering over an element. The appearance can be set by specifying the type of cursor, or by loading a specific icon. When using an icon, fallback images can be used, ending with a mandatory keyword fallback.

## Syntax

The `cursor` property can be set a specific `keyword` value:

```pseudo
cursor: keyword;
```

It can also accept one or more URLs:

```pseudo
cursor: url(cursor_image), url(fallback_cursor_image), keyword;
```

If the primary `cursor_image` fails to load, an optional `fallback_cursor_image` can be applied instead. However, in case all the image URLs fail to load, the mandatory `keyword` will be used as a last resort.

These are some common keywords used to specify the type of cursor to display:

| Keyword       |                                                   Example Cursor                                                   | Description                                                  |
| ------------- | :----------------------------------------------------------------------------------------------------------------: | ------------------------------------------------------------ |

> | auto          |                                                                                                                    | Detrmines the cursor to display based on context.            |

Reviewer 3: ```suggestion
| auto          |                                                                                                                    | Determines the cursor to display based on context.            |
```

| wait          |          ![Wait Cursor](https://raw.githubusercontent.com/Codecademy/docs/main/media/css-cursor-wait.png)          | System is busy and interaction is not allowed.               |
| cell          |          ![Cell Cursor](https://raw.githubusercontent.com/Codecademy/docs/main/media/css-cursor-cell.png)          | Indicates a table cell or cells can be selected.             |
| crosshair     |     ![Crosshair Cursor](https://raw.githubusercontent.com/Codecademy/docs/main/media/css-cursor-crosshair.png)     | Used to mark selections in an image.                         |
| text          |          ![Text Cursor](https://raw.githubusercontent.com/Codecademy/docs/main/media/css-cursor-text.png)          | Used to indicate editable text.                              |
| vertical-text | ![Vertical Text Cursor](https://raw.githubusercontent.com/Codecademy/docs/main/media/css-cursor-vertical-text.png) | Used to indicate editable vertical text.                     |

## Example

> In the example below, a `pointer` hand will appear whenever the cursor is hovered over the [paragraph](https://www.codecademy.com/resources/docs/html/paragraphs):

Reviewer 2: ```suggestion
In the example below, a `pointer` hand will appear whenever the cursor hovers over the [paragraph](https://www.codecademy.com/resources/docs/html/paragraphs):
```
